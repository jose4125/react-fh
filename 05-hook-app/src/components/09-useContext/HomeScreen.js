import { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  const { user } = useContext(UserContext);
  return (
    <div>
      <h1>home</h1>
      <pre>{JSON.stringify(user, null, 2)}</pre>
    </div>
  );
};
