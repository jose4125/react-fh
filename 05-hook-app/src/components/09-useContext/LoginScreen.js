import { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);
  return (
    <div>
      <h1>login</h1>
      <hr />
      <button onClick={() => setUser({ id: 2, name: "jose" })}>Login</button>
    </div>
  );
};
