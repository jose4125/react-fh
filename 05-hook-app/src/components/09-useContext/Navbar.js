import { NavLink } from "react-router-dom";

export const NavBar = () => {
  return (
    <nav>
      <ul>
        <li>
          <NavLink exact activeClassName="active" to="/">
            Home
          </NavLink>
        </li>
        <li>
          <NavLink activeClassName="active" to="/about">
            About
          </NavLink>
        </li>
        <li>
          <NavLink activeClassName="active" to="/login">
            Login
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};
