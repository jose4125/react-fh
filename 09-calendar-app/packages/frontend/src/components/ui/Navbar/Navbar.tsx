import { ReactElement } from 'react';

export const Navbar = (): ReactElement => {
  return (
    <div className="navbar navbar-dark bg-dark mb-4">
      <span className="navbar-brand">Pedro</span>
      <button className="btn btn-outline-danger">
        <i className="fas fa-sign-out-alt" /> <span>salir</span>
      </button>
    </div>
  );
};
