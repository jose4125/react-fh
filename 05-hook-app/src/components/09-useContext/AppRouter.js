import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./Navbar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <Switch>
          <Route exact path="/" component={HomeScreen} />
          <Route path="/about" component={AboutScreen} />
          <Route path="/login" component={LoginScreen} />
          <Redirect to="/" />
        </Switch>
      </div>
    </Router>
  );
};
