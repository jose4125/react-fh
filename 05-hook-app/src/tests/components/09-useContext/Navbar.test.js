import { shallow } from "enzyme";
import { NavBar } from "../../../components/09-useContext/Navbar";

describe("test NavBar", () => {
  const wrapper = shallow(<NavBar />);

  test("Should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have NavLinks", () => {
    expect(wrapper.find("NavLink").exists()).toBeTruthy();
    expect(wrapper.find("NavLink")).toHaveLength(3);
  });
});
