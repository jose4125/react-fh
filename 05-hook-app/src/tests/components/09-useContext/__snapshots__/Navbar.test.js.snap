// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test NavBar Should render properly 1`] = `
<nav>
  <ul>
    <li>
      <NavLink
        activeClassName="active"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="active"
        to="/about"
      >
        About
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="active"
        to="/login"
      >
        Login
      </NavLink>
    </li>
  </ul>
</nav>
`;
